src/misc/cross.cpp

```diff
@@ -16,26 +16,108 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
 
-
-#include "dosbox.h"
 #include "cross.h"
-#include "support.h"
+
 #include <string>
+#include <vector>
+
 #include <limits.h>
 #include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
 
 #ifdef WIN32
 #ifndef _WIN32_IE
 #define _WIN32_IE 0x0400
 #endif
 #include <shlobj.h>
+#else
+#include <libgen.h>
 #endif
 
-#if defined HAVE_SYS_TYPES_H && defined HAVE_PWD_H
-#include <sys/types.h>
+#if defined HAVE_PWD_H
 #include <pwd.h>
 #endif
 
+#include "support.h"
+
+#ifndef WIN32
+
+std::string cached_conf_path;
+
+static bool CreateDirectories(const std::string &path)
+{
+	struct stat sb;
+	if (stat(path.c_str(), &sb) == 0) {
+		const bool is_dir = ((sb.st_mode & S_IFMT) == S_IFDIR);
+		return is_dir;
+	}
+
+	std::vector<char> tmp(path.length());
+	std::copy(path.begin(), path.end(), tmp.begin());
+	std::string dname = dirname(tmp.data());
```

Comment:
Cool! vector's `data()` isn't `const` so can be fed into `dirname` :-)

Can use ranged-construction for the vector (also only one internal allocation similar to per-defining the size):
```
std::vector<char> tmp(path.begin(), path.end());
const std::string dname = dirname(tmp.data());
```

Reply:
Even better :) Thanks, updated implementation.